### PR TITLE
docs: document new argument glob for add command

### PR DESCRIPTION
**Pull request by ju0gri**
Fixes: #1928

Signed-off-by: Ioana Grigoropol <ioana.grigoropol@gmail.com>

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/command-reference/add.md [hunk 1]
```diff
@@ -144,6 +144,11 @@ directory symlinks.
 - `--external` - allow `targets` that are outside of the DVC repository. See
   [Managing External Data](/doc/user-guide/managing-external-data).
 
+- `--glob` - searches for pathnames matching the patterns specified by the
```
Comment (shcheklein) [inline content/docs/command-reference/add.md:147]: searches -> adds files ... (this is the primary function of the `dvc add` command)

File: content/docs/command-reference/add.md [hunk 2]
```diff
@@ -144,6 +144,11 @@ directory symlinks.
...
+  `targets`, according to rules used by the shell. It supports the Unix
```
Comment (shcheklein) [inline content/docs/command-reference/add.md:148]: `according to rules used by the shell`

`It supports the Unix shell-style wildcards`

sounds a bit repetitive

File: content/docs/command-reference/add.md [hunk 3]
```diff
@@ -144,6 +144,11 @@ directory symlinks.
...
+  shell-style wildcards `*`, `?`, `[seq]`, `[!seq]`, and the recursive glob
+  `**`. The order of the results depends on the file system.
```
Comment (shcheklein) [inline content/docs/command-reference/add.md:150]: what is the order of results from the `dvc add` perspective?

**Review by shcheklein — CHANGES_REQUESTED** [review]
Thanks @ju0gri for the PR 🙏 

It looks good and I can understand the idea, but I think we should go a bit higher level with the language, I left a few comments. PTAL when you have time.

File: content/docs/command-reference/add.md (hunk 1 above)
Comment (shcheklein) [reply content/docs/command-reference/add.md:147]: the idea here that people should be able to find it ... I would imagine the would look like (dvc add files matching patterns ... add directories matching ... etc)

File: content/docs/command-reference/add.md (hunk 3 above)
Comment (ju0gri) [reply content/docs/command-reference/add.md:150]: yeah, it makes no sense. I was focusing on the functionality behind it. I will correct it.

File: content/docs/command-reference/add.md (hunk 1 above)
Comment (ju0gri) [reply content/docs/command-reference/add.md]: Yes. I agree. I have updated it to better fit the functionality of dvc.

File: content/docs/command-reference/add.md [hunk 4]
```diff
@@ -144,6 +144,9 @@ directory symlinks.
 - `--external` - allow `targets` that are outside of the DVC repository. See
   [Managing External Data](/doc/user-guide/managing-external-data).
 
+- `--glob` - allows adding `targets` that match the specified pattern.
```
Comment (shcheklein) [inline content/docs/command-reference/add.md:147]: almost perfect! - one small suggestion to improve search- "allows adding files and directories that .... as specified by `target`"

 or `targets` can become part of the second sentence

File: content/docs/command-reference/add.md [hunk 5]
```diff
@@ -144,6 +144,9 @@ directory symlinks.
...
+  Shell-style wildcards are supported: `*`, `?`, `[seq]`, `[!seq]`, and `**`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/add.md:148]: How about
```suggestion
- `--glob` - enables `targets` to include path patterns. Wildcards supported:
  `*`, `?`, `[seq]`, `[!seq]`, `**`, etc. See
  [Unix style pathname pattern expansion](https://docs.python.org/3/library/glob.html)
  for details.
```

Comment (jorgeorpinel) [reply content/docs/command-reference/add.md:148]: Sorry guys, I realize you have already worked this and I just came with a different suggestion based on https://github.com/iterative/dvc/pull/4864#pullrequestreview-528893719. Please ignore this one for now, I'll apply it later if needed.

**Review by jorgeorpinel — APPROVED** [review]

File: content/docs/command-reference/add.md (hunk 4 above)
Comment (ju0gri) [reply content/docs/command-reference/add.md:147]: Sorry, I missed this comment. Updating it now.

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
Restyled version got merged. Thanks @ju0gri !

**Pull request closed without merging**